aip/general/0236.md

```diff
@@ -0,0 +1,297 @@
+- All resources under the experiments collection **must** be cascade-deleted if the top-level resource is deleted.
+- The etag on the experiment **should** guidance on Resource freshness validation ([AIP-154][aip-154]) and will be a randomly generated UUID ([AIP-148][aip-148]). The etag **should** be present in the preview logs.
+  - The parent resource **should** have an etag as well.
+  - Both etags **should** be present in the preview logs to uniquely identify the version of the live and experimental policies. These will change when either Policy configurations change.
+
+[aip-148]: https://aip.dev/148
+[aip-154]: https://aip.dev/154
+
+### Metadata
+A Metadata object is used to track all variables pertinent to a preview.
+ 
+```proto 
+message PreviewMetadata {
+  // State of the previewed resource. Possible values are ACTIVE and INACTIVE, the 
+  // former which indicates that results are being logged downstream.
+  string state = 1;
+
+  // Metadata string common to all preview mode logs for the product. Searching 
+  // logs for this string will isolate all logs pertaining to preview mode only.
+  string log_type = 2;
+
+  // The start time of the current preview.
+  google.protobuf.Timestamp start_time = 3;
+
+  // The latest stop time of the preview.
+  google.protobuf.Timestamp stop_time = 4;
+}
+```
+- PreviewMetadata **must** have the fields defined in the proto above and **must** use the same set of values for state.
+- A shared proto used across services **must not** be used ([AIP-213][aip-123]).
+- When a preview is first started, PreviewMetadata **must** be empty. It will only be present on experiments once a custom verb has been used to start it.
+- All fields **must** be output only.
+- The state field **must** only be updated by the Control Plane:
+  - The state **should** subsequently be altered between ACTIVE and INACTIVE to start or stop downstream evaluation using custom verbs described below.
+- The first time the startPreview custom verb is used, the Control Plane **must** create the PreviewMetadata and adds: 
+  - state = ACTIVE
+  - log_type string which is common to all preview mode logs for the product.
+  - start_time
+- When the stopPreview custom verb is used, the Control Plane **must** update PreviewMetadata with
+  - state = INACTIVE
+  - stop_time
+- start_time **must** be recreated every time the status is changed to ACTIVE.
+
+[aip-123]: https://aip.dev/123
+
+### Extensibility to Collection use-case
+
+For the Collection use-case, Rule is a repeated nested message on a Policy resource as shown below.
+
+```proto
+message Policy {
+
+  string name = 1;
+
+  Rule[] rules = 2;
+
+  // Other existing fields…
+}
+```
+
+A experiment is created for the entire parent resource. So essentially, there will be a experiment with its own set of rules. The following guidelines **should** be used to determine how to author an experiment when “rules” is a repeated nested message.
```

Comment:
Define "parent resource" at the very beginning of this AIP. It is not clear what this means. I know you mean that it is the resource under test, but I don't know that it is clear to the reader.